File: app/src/main/java/com/pandaq/pandamvp/framework/AppBaseFragment.java
package com.pandaq.pandamvp.framework;

import com.pandaq.appcore.framework.base.BaseFragment;
import com.pandaq.appcore.framework.base.BasePresenter;

/**
 * Created by huxinyu on 2018/1/26.
 * Email : panda.h@foxmail.com
 * Description :Fragment 基类
 */

public abstract class AppBaseFragment<P extends BasePresenter> extends BaseFragment<P> {

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/pandaq/pandamvp/template/AppBaseFragment.kt
package com.pandaq.pandamvp.template

import com.pandaq.appcore.framework.app.ActivityTask
import com.pandaq.appcore.framework.mvp.BaseFragment
import com.pandaq.uires.loading.LoadingDialogUtil
import com.pandaq.uires.msgwindow.Toaster

/**
 * Created by huxinyu on 2019/7/8.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :
 */
abstract class AppBaseFragment<P : AppBasePresenter<*>> : BaseFragment<P>() {
    override fun showLoading(msg: String?) {
        LoadingDialogUtil.show(ActivityTask.getInstance().currentActivity(), msg)
    }

    override fun showLoading() {
        showLoading(true)
    }

    override fun showLoading(cancelAble: Boolean) {
        LoadingDialogUtil.show(ActivityTask.getInstance().currentActivity(), cancelAble)
    }

    open fun hideLoading() {
        LoadingDialogUtil.hideProgressQuick()
    }

    override fun onError(errCode: Long, errMsg: String?) {
        errMsg?.let {
            Toaster.showError(errMsg)
        }
    }

    override fun onFinish(success: Boolean) {
        hideLoading()
    }
}